### PR TITLE
feat: Export defineConfig defines the auxiliary function of the configuration

### PR DESCRIPTION
**Pull request by rxliuli**
This PR contains:
- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [ ] yes (*bugfixes and features will not be merged without tests*)
- [x] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers: https://github.com/rollup/rollup/issues/4125

### CODE REVIEW

**Comment by rxliuli** [comment]
@lukastaegert I added some documentation at the end

File: docs/01-command-line-reference.md [hunk 1]
```diff
@@ -200,6 +200,33 @@ export default commandLineArgs => {
 }
 ```
 
+Config Intellisense
```
Comment (lukastaegert) [inline docs/01-command-line-reference.md:203]: You should make this a `####` sub-heading so that it looks right on the web site

Comment (lukastaegert) [reply docs/01-command-line-reference.md:203]: Otherwise looks great!

Comment (rxliuli) [reply docs/01-command-line-reference.md]: added: https://github.com/rollup/rollup/pull/4127/commits/555cac31aa8cbf61c3a17b4c47e33bde720bfb5a

**Review by lukastaegert — APPROVED** [review]
Thanks 👍

**Comment by rxliuli** [comment]
I would like to ask, if yes, is it possible to support ts type configuration files? If I create a new PR to support this feature, will it be merged?

**Comment by lukastaegert** [comment]
There is already a PR for this which got stalled: #3835
However, discussions were already pretty advanced, maybe we can move that one forward? The general idea would be to allow using plugins to transpile the config file, just the specifics were not finalized.

**Comment by rxliuli** [comment]
> There is already a PR for this which got stalled: #3835
> However, discussions were already pretty advanced, maybe we can move that one forward? The general idea would be to allow using plugins to transpile the config file, just the specifics were not finalized.

It has been 8 months and it has not been confirmed yet. . . In my opinion, this is a lesson learned, just "copy homework".

- gulp: Use ts-node to load ts configuration files
- vite: package vite.config.ts => vite.config.ts.js, then load the js file

**Pull request merged**